### PR TITLE
MAINT: renaming elasticsearch to opensearch -- second pass

### PR DESCRIPTION
**Pull request by chenqi0805**
### Description
This PR

1. renames es version to opensearch version
2. fixes hyperlinks to redirect to opensearch-project
3. rename elasticsearch into opensearch wherever proper in .md
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java [hunk 1]
```diff
@@ -107,7 +107,7 @@ public void testPipelineEndToEnd() throws InterruptedException {
         sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_2, exportTraceServiceRequestTrace2BatchWithRoot);
         sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_2, exportTraceServiceRequestTrace1BatchNoRoot);
 
-        //Verify data in elasticsearch sink
+        //Verify data in elasticsearch backend
```
Comment (wrijeff) [inline data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java:110]: Replace elasticsearch w/ opensearch here too?

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java [hunk 2]
```diff
@@ -76,7 +76,7 @@ public void testPipelineEndToEnd() throws IOException, InterruptedException {
         sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_1, exportTraceServiceRequest11);
         sendExportTraceServiceRequestToSource(DATA_PREPPER_PORT_2, exportTraceServiceRequest12);
 
-        //Verify data in elasticsearch sink
+        //Verify data in elasticsearch backend
```
Comment (wrijeff) [inline data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java:79]: Similar comment

File: data-prepper-plugins/opensearch/README.md [hunk 3]
```diff
@@ -116,7 +116,7 @@ Default value is false. Set it to true for [Service map trace analytics](#servic
 
 - <a name="template_file"></a>`template_file`(optional): A json file path to be read as index template for custom data ingestion. The json file content should be the json value of
 `"template"` key in the json content of elasticsearch [Index templates API](https://www.elastic.co/guide/en/elasticsearch/reference/7.8/index-templates.html), 
-e.g. [otel-v1-apm-span-index-template.json](https://github.com/opendistro-for-elasticsearch/data-prepper/blob/main/data-prepper-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json)
+e.g. [otel-v1-apm-span-index-template.json](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/elasticsearch/src/main/resources/otel-v1-apm-span-index-template.json)
```
Comment (wrijeff) [inline data-prepper-plugins/opensearch/README.md:119]: URL still has elasticsearch, 404's. Should be https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-plugins/opensearch/src/main/resources/otel-v1-apm-span-index-template.json

File: data-prepper-plugins/opensearch/security.md [hunk 4]
```diff
@@ -42,13 +42,13 @@ Please check this [doc](https://docs.aws.amazon.com/elasticsearch-service/latest
...
-The Elasticsearch sink creates an [Index State Management (ISM)](https://opendistro.github.io/for-elasticsearch-docs/docs/ism/) policy for Trace Analytics indices but Amazon Elasticsearch Service allows only the `master user` to create an ISM policy. So,
+The OpenSearch sink creates an [Index State Management (ISM)](https://opendistro.github.io/for-elasticsearch-docs/docs/ism/) policy for Trace Analytics indices but Amazon Elasticsearch Service allows only the `master user` to create an ISM policy. So,
  
  * If you use IAM for your master user in FGAC domain, configure the sink as below,
   
   ```
   sink:
-      elasticsearch:
+      opensearch:
         hosts: ["https://your-fgac-amazon-elasticssearch-service-endpoint"]
```
Comment (wrijeff) [inline data-prepper-plugins/opensearch/security.md:52]: Wondering if this should be updated too, even though there isn't an Amazon hosted OpenSearch service

File: docs/readme/trace_setup.md [hunk 5]
```diff
@@ -121,7 +121,7 @@ service-map-pipeline:
 
 ### Sink Setup
 
-In the above trace analytics configuration yaml file, we use elasticsearch as the sink. This sink could be Amazon Elasticsearch domain or Opendistro for Elasticsearch. You can configure this according to your setup. Please check the [Data Prepper Elasticsearch Sink](../../data-prepper-plugins/elasticsearch/README.md#Configuration) for more details. 
+In the above trace analytics configuration yaml file, we use elasticsearch as the sink. This sink could be Amazon Elasticsearch domain or Opendistro for Elasticsearch. You can configure this according to your setup. Please check the [Data Prepper OpenSearch Sink](../../data-prepper-plugins/opensearch/README.md#Configuration) for more details.
```
Comment (wrijeff) [inline docs/readme/trace_setup.md:124]: >This sink could be...

Should reference OpenSearch here

**Review by wrijeff — APPROVED** [review]
Thanks for the maintenance tasks 👍

File: data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java (hunk 1 above)
Comment (chenqi0805) [reply data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java:110]: It is still using ODFE docker backend so I did not rename it yet

File: data-prepper-plugins/opensearch/security.md (hunk 4 above)
Comment (chenqi0805) [reply data-prepper-plugins/opensearch/security.md:52]: Tend to push back that change.

**Pull request merged**